controllers/chat/getSignedChannelList.js

```diff
@@ -62,7 +84,7 @@ const getSignedChannelList = async (req, res) => {
       queriedChannels.push(...response);
     });
 
-    const channels = queriedChannels.map(__transformChannelData);
+    const channels = queriedChannels.reduce(__filterAndTransformChannelData, []);
     return res.status(200).json(responseSuccess('Success retrieve channels', channels));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from mapping over `queriedChannels` to reducing them is a logical change that aligns with the modification made to `__filterAndTransformChannelData`. The reduce function filters and transforms the channel data in one go, which could improve performance by avoiding an additional iteration over the channels. This change seems fine as long as it aligns with the application's requirements.


<!-- This is an auto-generated comment by OSS CodeRabbit -->